r14423 | batrick | 2009-07-18 14:56:07 +0000 (Sat, 18 Jul 2009) | 6 lines

If neither --script nor -sV is present, then do not load NSE. Before, NSE
would load despite script scanning not being activated. See [1] for
preliminary patch from Solar Designer and motivation.

[1] http://seclists.org/nmap-dev/2009/q3/0207.html

Index: nmap.cc
===================================================================
--- nmap.cc	(revision 14422)
+++ nmap.cc	(revision 14423)
@@ -1681,7 +1681,8 @@
   }
   if (o.servicescan)
     o.scriptversion = 1;
-  open_nse();
+  if (o.scriptversion || o.script)
+    open_nse();
 #endif
   
   /* Time to create a hostgroup state object filled with all the requested
@@ -2017,7 +2018,8 @@
   if (o.extra_payload) free(o.extra_payload);
   if (o.ipoptions) free(o.ipoptions);
 #ifndef NOLUA
-  close_nse();
+  if (o.scriptversion || o.script)
+    close_nse();
   free(o.scriptargs);
 #endif
 }
